src/main/scala/sbtcrossproject/Platform.scala

```diff
@@ -0,0 +1,59 @@
+package sbtcrossproject
+
+/**
+ * Created by grinder on 17.07.16.
+ */
+abstract class Platform(val name: String, val path: String)
+
+object Shared extends Platform("shared", "shared/src/main") {
+
+  implicit class SharedTypeExt[T <: Shared.type](project: CrossProject[T]) {
+    def settings(settings: Seq[_root_.sbt.Def.Setting[_]]): CrossProject[T] =
```

Comment:
Using `_root_` is smelly. Have an `import sbt._` and use `Setting[_]`.

There are other occurrences of this code pattern.